### PR TITLE
Fix file deletion

### PR DESCRIPTION
**Pull request by almirsarajcic**
We couldn't delete records from the `FileIndex` because each message had a different key for the same file.
Now we use the same key which makes it possible to check for other instances of the same file (https://github.com/Buckitup-chat/chat/pull/202/files#diff-d938b8e0e8e238103c8e7fed6ff3493af3705628bad718c89060fa6142c58f3aR13).

Also, files couldn't be deleted from the dialogs at all because we were sending incorrect message ID (https://github.com/Buckitup-chat/chat/pull/202/files#diff-9fbc183bef396d8bacd542491a315bacb688c57e0c3ee5f7cdf16fc945d9493fR123 and https://github.com/Buckitup-chat/chat/pull/202/files#diff-9fbc183bef396d8bacd542491a315bacb688c57e0c3ee5f7cdf16fc945d9493fR135).
Besides that, content wasn't being deleted from My notes (https://github.com/Buckitup-chat/chat/pull/202/files#diff-9fbc183bef396d8bacd542491a315bacb688c57e0c3ee5f7cdf16fc945d9493fR157).

### CODE REVIEW

File: lib/chat_web/live/helpers/uploader.ex [hunk 1]
```diff
@@ -179,7 +179,7 @@ defmodule ChatWeb.LiveHelpers.Uploader do
       entry.client_last_modified
     ]
     |> Enum.join(":")
-    |> Enigma.hash()
+    |> Enigma.short_hash()
```
Comment (sergey-lukianov) [inline lib/chat_web/live/helpers/uploader.ex:182]: Should be just `Enigma.hash`

 `Base.encode16(case: :lower)` to get string
But use it on the edge of system. I.e when you need url or JSON

File: lib/chat_web/live/main_live/page/dialog.ex [hunk 2]
```diff
@@ -209,17 +209,6 @@ defmodule ChatWeb.MainLive.Page.Dialog do
     |> assign(:edit_message_id, nil)
   end
 
-  def delete_message(
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/page/dialog.ex:212]: why to delete this? we have text message removal AFAIR

File: lib/chat/content/files.ex [hunk 3]
```diff
@@ -10,10 +10,14 @@ defmodule Chat.Content.Files do
   def delete({key, _secret}), do: delete(key)
   def delete(key), do: Db.delete(db_key(key))
 
-  def add(data) do
-    key = UUID.uuid4()
+  def add([key, raw_secret, _, _, _, _] = data) do
+    secret = Base.decode64!(raw_secret)
```
Comment (sergey-lukianov) [inline lib/chat/content/files.ex:14]: Should this be `url_decode`?

File: lib/chat_web/live/main_live/page/dialog.ex (hunk 2 above)
Comment (almirsarajcic) [reply lib/chat_web/live/main_live/page/dialog.ex:212]: I've searched for the reference. I don't think we do a single message deletion anywhere in the dialog and the room.

File: lib/chat/content/files.ex (hunk 3 above)
Comment (almirsarajcic) [reply lib/chat/content/files.ex:14]: No, this is the same as this:
https://github.com/Buckitup-chat/chat/blob/77980292a8b0362c37edab76df1b42088cd441c3/lib/chat_web/controllers/file_controller.ex#L40-L44

**Review by sergey-lukianov — APPROVED** [review]

**Pull request merged**